src/NServiceBus.NHibernate/SharedSession/SharedSessionConfig.cs

```diff
@@ -0,0 +1,21 @@
+namespace NServiceBus.Persistence.NHibernate
+{
+    using Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+
+    /// <summary>
+    /// Deduplication configuration extensions.
+    /// </summary>
+    public static class SharedSessionConfig
+    {
+        /// <summary>
+        /// Disables sharing a connection between SQL Server transport and NHibernate persistence. Intended to be used in conjunction with Outbox only.
+        /// </summary>
+        /// <param name="persistenceConfiguration"></param>
+        public static PersistenceExtentions<NHibernatePersistence> DoNotShareTransportConnection(this PersistenceExtentions<NHibernatePersistence> persistenceConfiguration)
```

Comment:
That is not the intent @SimonCropp.
The chaining in this API is important and there is no order.
In this API the user can do things like:
```c#
config.UsePersistence<NHibernate, StorageType.Outbox>()
                                     .DoNotShareTransportConnection()
                                     .ConnectionString(connString)
                                     .ToggleSomeOtherConfigForOutbox();
```
And the above are all the configuration settings for `NHibernate` and `StorageType.Outbox`.
Does this make sense ?